cmd/kubectl-gadget/seccompadvisor.go

```diff
@@ -0,0 +1,143 @@
+
+package main
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	"github.com/kinvolk/inspektor-gadget/cmd/kubectl-gadget/utils"
+	gadgetv1alpha1 "github.com/kinvolk/inspektor-gadget/pkg/api/v1alpha1"
+)
+
+var seccompAdvisorCmd = &cobra.Command{
+	Use:   "seccomp-advisor",
+	Short: "Generate seccomp policies based on recorded syscalls activity",
+}
+
+var seccompAdvisorStartCmd = &cobra.Command{
+	Use:   "start",
+	Short: "Start to monitor the system calls",
+	RunE:   runSeccompAdvisorStart,
+}
+
+var seccompAdvisorStopCmd = &cobra.Command{
+	Use:   "stop",
+	Short: "Stop monitoring and report the policies",
+	RunE:   runSeccompAdvisorStop,
+}
+
+var seccompAdvisorListCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List existing seccomp traces",
+	RunE:   runSeccompAdvisorList,
+}
+
+var (
+	traceOutput string
+	traceOutputMode string
+)
+
+func init() {
+	// Add generic information.
+	rootCmd.AddCommand(seccompAdvisorCmd)
+	utils.AddCommonFlags(seccompAdvisorCmd, &params)
+
+	seccompAdvisorCmd.AddCommand(seccompAdvisorStartCmd)
+	seccompAdvisorStartCmd.PersistentFlags().StringVarP(&traceOutputMode, "trace-output-mode", "m", "Status", "Output mode")
+	seccompAdvisorStartCmd.PersistentFlags().StringVar(&traceOutput, "trace-output", "", "Output")
+
+	seccompAdvisorCmd.AddCommand(seccompAdvisorStopCmd)
+	seccompAdvisorCmd.AddCommand(seccompAdvisorListCmd)
+}
+
+// runSeccompAdvisorStart starts monitoring of syscalls for the given
+// parameters.
+func runSeccompAdvisorStart(cmd *cobra.Command, args []string) error {
+	config := &utils.TraceConfig{
+		GadgetName:      "seccomp",
+		Operation:       "start",
+		TraceOutputMode: traceOutputMode,
+		TraceOutput:     traceOutput,
+		CommonFlags:     &params,
+	}
+
+	traceID, err := utils.CreateTrace(config)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("%s\n", traceID)
+
+	return nil
+}
+
+// runSeccompAdvisorStop reports an already running trace which ID was given
+// as parameter.
+func runSeccompAdvisorStop(cmd *cobra.Command, args []string) error {
+	if len(args) == 0 {
+		return errors.New("Usage: kubectl gadget seccomp-advisor stop global-trace-id\n")
+	}
+
+	callback := func(results []gadgetv1alpha1.Trace) error {
+		for _, i := range results {
+			fmt.Printf("%v\n", i.Status.Output)
+		}
+
+		return nil
+	}
+
+	traceID := args[0]
+
+	err := utils.SetTraceOperation(traceID, "generate")
+	if err != nil {
+		return err
+	}
+
+	// We stop the trace so its Status.State become Stopped.
+	// Indeed, generate operation does not change value of Status.State.
+	err = utils.SetTraceOperation(traceID, "stop")
+	if err != nil {
+		return err
+	}
+
+	err = utils.PrintTraceOutputFromStatus(traceID, "Stopped", callback)
+	if err != nil {
+		return err
+	}
+
+	// WARNING We should defer this call, but trace can sadly timeout, so it is
+	// better for the user to call this function again to get expected result and
+	// delete the trace at the same time.
+	return utils.DeleteTrace(traceID)
```

Comment:
I am aware about this problem, this is why I added the `if params.Podname == ""` in `*Start()`.
This is clearly bad, but until I find a way to reduce trace timeout, I think it is better to let it like this since user can still remove the trace with `kubectl delete trace -n gadget` instead of having the trace deleted before the user can get the result.
What do you think?

Reply:
Understood, I didn't know that was the motivation for having the podname check. 

Please re-add it so that we can go ahead with this PR. You can open an issue for the timeout and we can work on it for another PR.

Reply:
I will test again if the trace is removed when using `ExternalResource` without giving a podname, even though I am not really sure.
Otherwise I think I will add this test back.